connections/hole-punching.md

```diff
@@ -83,14 +90,13 @@ platform types.
 
 A transport protocol allows basic host-to-host communication. Examples are TCP,
 Websockets building on top of TCP, as well as the protocols layered over UDP
-namely QUIC and WebRTC.
+namely QUIC, [WebRTC] and [WebTransport].
```

Comment:
WebRTC DataChannels run on top of SCTP which runs on top of DTLS which runs on top of UDP.

Reply:
NOTE: WebRTC could run on top of TCP as well (it's just that the support for it is lacking in webrtc-rs due to no demand; not sure about other languages)

Reply:
I was unaware of that. Thanks @melekes. Updated in https://github.com/libp2p/specs/pull/422/commits/ec692b3824397178e6cdeb129957561a1a34dd85.

Also added this as an open question to the WebRTC specification pull request https://github.com/libp2p/specs/pull/412/commits/81249f3f103bbde165e78518e0d91139fbddb994. Curious what you think @melekes.

I guess long term it would be good to support both, e.g. in the scenario where a browser tries to connect to a public non-browser without TLS certificate, where the browsers NAT drops all UDP traffic.

Reply:
It would be good to support both indeed. The public non-browser node can listen on both TCP and UDP port and expose 2 addresses (`ip/A.B.C.D/udp/4444/webrtc` and `ip/A.B.C.D/tcp/3333/webrtc`). The browser can try UDP first and that fails fallback to TCP.